src/Microsoft.AspNet.Server.Kestrel/Http/DateHeaderValueManager.cs

```diff
@@ -57,103 +61,159 @@ public DateHeaderValueManager()
         /// <returns>The value.</returns>
         public virtual string GetDateHeaderValue()
         {
-            PumpTimer();
-
-            // See https://msdn.microsoft.com/en-us/library/az4se3k1(v=vs.110).aspx#RFC1123 for info on the format
-            // string used here.
-            // The null-coalesce here is to protect against returning null after Dispose() is called, at which
-            // point _dateValue will be null forever after.
-            return _dateValue ?? _systemClock.UtcNow.ToString(Constants.RFC1123DateFormat);
+            _hadRequestsSinceLastTimerTick = true;
+            PrepareDateValues();
+            return _dateValue;
```

Comment:
Yes, it will, however,  `PrepareDateValues()` will return the date from the real time clock if `_isDisposed` is true. The older code behaved this way too.